xlators/features/trash/src/trash.c

```
@@ -2501,7 +2532,17 @@ init(xlator_t *this)
         goto out;
     }
 
-    priv->trash_itable = inode_table_new(0, this, 0, 0);
+    if (priv->state) {
+        priv->trash_itable = inode_table_new(0, this, 0, 0);
+        if (!priv->trash_itable) {
+            ret = ENOMEM;
```

Shouldn't we return a negative error ?

---

done